MadeByMunstersLLC/angular-waiting-button
# Refactor use commonJS for npm 3 15 17

**Changes**:
Strips out Gulp and replaces it with Webpack for support with ES6 modules. In addition updates the Readme for updated examples and reworks the demo-app to use ES6 syntax module loading. Some notes:

- We are resolving angular through webpack so that it's not bundled in with the library
- I updated the name to be more import friendly as `mbm.waitingButton` was not. If you have a better solution for this let me know.
- added a `.npmignore` file to ignore the demo-app from the NPM package once added. This folder is mostly bloated by the inclusion of angular within the bundle and therefore shouldn't be needed within the npm install itself.
- update readme to the ES6 example within the demo-app folder.
- included style-loader to bundle in any css added.
- updates karma to use webpack as a build tool.
- includes test script for karma in NPM package.

**Testing**:
- pull down the branch and npm install 
  - note: maybe a good idea to remove node_modules with all the changes
- check test by running `npm run test`
- check bundling by running `npm run build`
- check demo-app by `cd demo-app` & `npm install` & `npm run server`
  - note the demo app loads the NPM package from the directory above it.

**Trello Links**:
https://trello.com/c/FR6ZvdhG/218-move-angular-waiting-button-to-es6-module-and-npm
